src/stream/Encryption.ts

```diff
@@ -28,36 +33,75 @@ class InvalidGroupKeyError extends Error {
     }
 }
 
-export class GroupKey {
+type GroupKeyObject = {
+    id: string,
+    hex: string,
+    data: Uint8Array,
+}
+
+type GroupKeyProps = {
+    groupKeyId: string,
+    groupKeyHex: string,
+    groupKeyData: Uint8Array,
+}
+
+function GroupKeyObjectFromProps(data: GroupKeyProps | GroupKeyObject) {
+    if ('groupKeyId' in data) {
+        return {
+            id: data.groupKeyId,
+            hex: data.groupKeyHex,
+            data: data.groupKeyData,
+        }
+    }
+
+    return data
+}
+
+interface GroupKey extends GroupKeyObject {}
+
+// eslint-disable-next-line no-redeclare
+class GroupKey {
     static InvalidGroupKeyError = InvalidGroupKeyError
 
-    static validate(maybeGroupKey) {
+    static validate(maybeGroupKey: GroupKey) {
         if (!maybeGroupKey) {
-            throw new InvalidGroupKeyError(`value must be a ${this.name}: ${util.inspect(maybeGroupKey)}`)
+            throw new InvalidGroupKeyError(`value must be a ${this.name}: ${util.inspect(maybeGroupKey)}`, maybeGroupKey)
         }
 
         if (!(maybeGroupKey instanceof this)) {
-            throw new InvalidGroupKeyError(`value must be a ${this.name}: ${util.inspect(maybeGroupKey)}`)
+            throw new InvalidGroupKeyError(`value must be a ${this.name}: ${util.inspect(maybeGroupKey)}`, maybeGroupKey)
         }
 
         if (!maybeGroupKey.id || typeof maybeGroupKey.id !== 'string') {
-            throw new InvalidGroupKeyError(`${this.name} id must be a string: ${util.inspect(maybeGroupKey)}`)
+            throw new InvalidGroupKeyError(`${this.name} id must be a string: ${util.inspect(maybeGroupKey)}`, maybeGroupKey)
+        }
+
+        if (maybeGroupKey.id.includes('---BEGIN')) {
+            throw new InvalidGroupKeyError(
+                `${this.name} public/private key is not a valid group key id: ${util.inspect(maybeGroupKey)}`,
+                maybeGroupKey
+            )
         }
 
         if (!maybeGroupKey.data || !Buffer.isBuffer(maybeGroupKey.data)) {
-            throw new InvalidGroupKeyError(`${this.name} data must be a buffer: ${util.inspect(maybeGroupKey)}`)
+            throw new InvalidGroupKeyError(`${this.name} data must be a Buffer: ${util.inspect(maybeGroupKey)}`, maybeGroupKey)
         }
 
         if (!maybeGroupKey.hex || typeof maybeGroupKey.hex !== 'string') {
-            throw new InvalidGroupKeyError(`${this.name} hex must be a string: ${util.inspect(maybeGroupKey)}`)
+            throw new InvalidGroupKeyError(`${this.name} hex must be a string: ${util.inspect(maybeGroupKey)}`, maybeGroupKey)
         }
 
         if (maybeGroupKey.data.length !== 32) {
-            throw new InvalidGroupKeyError(`Group key must have a size of 256 bits, not ${maybeGroupKey.data.length * 8}`)
+            throw new InvalidGroupKeyError(`Group key must have a size of 256 bits, not ${maybeGroupKey.data.length * 8}`, maybeGroupKey)
         }
+
     }
 
-    constructor(groupKeyId, groupKeyBufferOrHexString) {
```

Comment:
Maybe move this class to a separate class? It would help the code navigation, if we could just open `GroupKey.ts` file to check the implementation of the class. 

It would make sense to move some other classes to separate files, too. Maybe in another PR?

Reply:
👍 sure, let's get these PRs merged first though